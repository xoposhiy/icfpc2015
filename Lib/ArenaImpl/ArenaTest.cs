using System.IO;
using ApprovalTests;
using Lib.Intelligence;
using Newtonsoft.Json;
using NUnit.Framework;

namespace Lib.ArenaImpl
{
    [TestFixture]
    public class ArenaTest
    {
        [Test]
        public void Test()
        {
            var arena = new Arena();
//            var solver = new PhrasesOnlySolver();
//            var solver = new NamiraOracle();
            var solver = new AzuraOracle();
            var res = arena.RunAllProblems(solver);
//            File.WriteAllText("arena.json", JsonConvert.SerializeObject(res, Formatting.Indented));
            Approvals.Verify(res);
        }
    }
}