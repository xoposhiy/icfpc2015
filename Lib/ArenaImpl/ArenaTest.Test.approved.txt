TotalScore: 515
ProblemId	AvgScore	ScoreBySeeds
0	15	15
1	12	12
2	27	25,25,20,25,20,25,25,35,25,45
3	15	16,13,19,13,16
4	16	20,24,20,12,16,16,16,20,16,16,28,12,12,12,12,16,16,12,16,12,12,32,24,16,20,12,16,20,12,12,16,20,20,20,16,16,16,12,12,16,12,12,12,16,24,20,20,28,20,12
5	16	12,17,17,14,14,20,13,16,20,18
6	12	23,30,5,4,24,32,4,4,4,30,21,25,4,4,4,4,4,4,4,24,4,4,4,4,4,22,4,8,22,21,24,4,29,12,4,28,16,4,18,8,7,10,20,4,5,21,4,36,4,30
7	14	4,4,17,26,21
8	21	28,23,11,23,28,25,13,28,27,11
9	7	8,7,7,7,7
10	2	2
11	19	26,8,19,22,20
12	49	63,57,49,54,51,39,50,60,41,35
13	34	34
14	8	8
15	24	24
16	17	17
17	128	128
18	17	17
19	33	33
20	16	16
21	3	3
22	4	4
23	6	6